spring-boot-admin-server-ui-javamelody/src/main/java/spring/boot/admin/javamelody/configuration/JavamelodyAutoConfiguration.java

```diff
@@ -0,0 +1,44 @@
+package spring.boot.admin.javamelody.configuration;
+
+import net.bull.javamelody.CollectorServlet;
+import net.bull.javamelody.JavaMelodyService;
+import net.bull.javamelody.MonitoringFilter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.boot.web.servlet.ServletRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import javax.servlet.Filter;
+
+// Always run, if you depend on this module, you will get this autoconfiguration
+@Configuration
+@ConditionalOnProperty(prefix = "javamelody", name = "collectserver.enabled", matchIfMissing = true)
+public class JavamelodyAutoConfiguration {
+
+    private static final String CONTEXT_ROOT = "/javamelody";
+
+    @Bean
+    public JavaMelodyService javaMelodyService() {
+        return new JavaMelodyService();
+    }
+
+    @Bean
+    public FilterRegistrationBean filterRegistrationBean() {
+        final Filter javaMelodyFilter = new MonitoringFilter();
+
+        FilterRegistrationBean javaMelodyFilterBean = new FilterRegistrationBean(javaMelodyFilter);
+        javaMelodyFilterBean.addServletNames("monitoring");
+        javaMelodyFilterBean.addUrlPatterns("/*");
+        return javaMelodyFilterBean;
+    }
```

Comment:
The idea was to monitor spring boot admin itself. Why is this not enough?

Reply:
The FilterRegistrationBean is not enough because in general a SessionListener is also installed. See https://github.com/javamelody/javamelody/blob/master/javamelody-spring-boot-starter/src/main/java/net/bull/javamelody/JavaMelodyAutoConfiguration.java#L87
Then the spring-boot-starter also adds monitoring of RestControllers, RestTemplates, Services, spring context and more.
So adding the [javamelody-spring-boot-starter](https://github.com/javamelody/javamelody/wiki/SpringBootStarter) to monitor the spring-boot-admin server should be much simpler than adding filter, sessionlistener, etc.